src/test/java/hu/elte/eserial/model/SetterTest.java

```diff
@@ -0,0 +1,95 @@
+package hu.elte.eserial.model;
+
+import hu.elte.eserial.exception.EserialInvalidMethodException;
+import org.junit.Test;
+
+import java.lang.reflect.Method;
+import java.lang.reflect.Type;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+
+public class SetterTest {
+    public class ClassWithSetters {
+        private int id;
+
+        public void setId(int id) {
+            this.id = id;
+        }
+
+        public void invalidSet(int id) {
+            this.id = id;
+        }
+
+        void setterThatThrows(int id) throws Exception {
+            throw new Exception();
+        }
+
+        public int getId() {
+            return id;
+        }
+    }
+
+    @Test(expected = EserialInvalidMethodException.class)
+    public void getElementName_GivenAnInValidSetter_ThrowsEserialInvalidMethodException() {
+        try {
+            Method method = ClassWithSetters.class.getDeclaredMethod("invalidSet", int.class);
+
+            Setter setter = new Setter(new ClassWithSetters(), method);
+            setter.getElementName();
+        } catch (NoSuchMethodException e) {
+            fail(e.getMessage());
```

Comment:
Dont't do this, it's confusing. Just add `throws NoSuchMethodException` to the method name.